packages/select/src/components/multi-select/multiSelect.tsx

```diff
@@ -346,12 +365,24 @@ export class MultiSelect<T> extends AbstractPureComponent<MultiSelectProps<T>, M
 
             this.props.popoverProps?.onInteraction?.(nextOpenState, evt);
         });
+    };
 
     private handlePopoverOpened = (node: HTMLElement) => {
         if (this.queryList != null) {
             // scroll active item into view after popover transition completes and all dimensions are stable.
             this.queryList.scrollActiveItemIntoView();
         }
+
+        // Unless user explicitly sets autoFocus to false, for customTargets we should by default
+        // focus on the input when the Popover opens.
```

Comment:
nit: I find it better to pull out to better names so it reads like what you would comment, ex
```
const hasCustomTarget = this.props.customTarget != null;
if (hasCustomTarget && this.input != null) {
  const shouldAutofocus = this.props.tagInputProps?.inputProps?.autoFocus !== false;
  if (shouldAutofocus) {
    this.input.focus();
  }
}
```
personally I don't think `this.input != null` needs anything since that's just a ref guard that we always expect to be set by the time this is called

code comments risk getting out of sync with the code, while pulling out smaller variables allows each step to be grokked as you go